tests/acceptance/nested-preprocessor-test-slow.js

```diff
@@ -0,0 +1,332 @@
+'use strict';
+
+var path = require('path');
+var fs = require('fs-extra');
+
+var AppFixture = require('../helpers/app-fixture');
+var AddonFixture = require('../helpers/addon-fixture');
+var InRepoAddonFixture = require('../helpers/in-repo-addon-fixture');
+
+var processTemplate = require('../../lib/utilities/process-template');
+
+var root = path.resolve(__dirname, '..', '..');
+var CommandGenerator = require('../../tests/helpers/command-generator');
+var ember = new CommandGenerator(path.join(root, 'bin', 'ember'));
+
+var chai = require('../chai');
+var expect = chai.expect;
+var file = chai.file;
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var setupPreprocessorRegistryFixture = function(selfOrParent, registry) {
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  registry.add('<%= registryType %>', {
+    name: addon.name + '-<%= registryType %>',
+    ext: '<%= ext %>',
+    toTree: function(tree) {
+      return stew.map(tree, function(content, relativePath) {
+        var marker = '// ' + addon.name + '-<%= registryType %>-preprocessor-transform-' + selfOrParent + ' /' + relativePath;
+        addon._findHost().project.ui.writeLine(marker);
+        return marker + '\n' + content;
+      });
+    }
+  });
+};
+function generatePreprocessor(context) {
+  return processTemplate(setupPreprocessorRegistryFixture.toString(), context);
+}
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var preprocessTreeFixture = function(type, tree) {
+  if (type === 'all') { return tree; }
+
+  // THIS IS A WORKAROUND FOR https://github.com/ember-cli/ember-cli/issues/6512
+  if (type === 'css' && tree.toString().indexOf('Concat: Vendor Styles') !== -1) {
+    return tree;
+  }
+
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  // We're going to add a marker. The `postprocessTree` function will remove
+  // the marker if present and identify whether or not it found the marker.
+  var marker = '// ' + addon.name + '-preprocessTree(' + type + ')' + ' /';
+
+  tree = stew.map(tree, function(content, relativePath) {
+    var localMarker = marker + relativePath;
+    addon._findHost().project.ui.writeLine(localMarker);
+    return localMarker + '\n' + content;
+  });
+
+  return tree;
+};
+
+// THIS IS A FIXTURE. It also happens to be valid JavaScript.
+var postprocessTreeFixture = function(type, tree) {
+  if (type === 'all') { return tree; }
+
+  var stew = require('broccoli-stew');
+  var addon = this;
+
+  var preprocessTreeMarker = '// ' + addon.name + '-preprocessTree(' + type + ')';
+  var marker = '// ' + addon.name + '-postprocessTree(' + type + ')';
+
+  // We're going to inspect state and add an appropriate marker.
+  tree = stew.map(tree, function(content, relativePath) {
+
+    // Skip assets we know that the CSS preprocessor adds.
+    var cssName = addon._findHost().name;
```

Comment:
[`project.name()` is not treated as the canonical value inside of `ember-app`.](https://github.com/ember-cli/ember-cli/blob/e64c27a651295af8cd1da907eb55fd8f35c0de6f/lib/broccoli/ember-app.js#L112) It's only used for `isDevelopingAddon` and inside of the constructor. I'd rather stop one short of the project and rely on it to tell me what name will eventually be used.